lab_2_tokenize_by_bpe/main.py

```diff
@@ -14,6 +14,18 @@ def prepare_word(
     :param end_of_word: a token that signifies the end of word
     :return: preprocessed word
     """
+    if (not isinstance(raw_word, str) or not isinstance(start_of_word, str | None)
+            or not isinstance(end_of_word, str | None)):
```

Comment:
I believe isinstance is not a good way to check if a value is `None`. As it is a singleton, `is` is the appropriate operator.